File: swift/MagCry/Views/BotQuoteView.swift
import SwiftUI

/// Shows a single bot's live quote with Buy / Sell / Pass action buttons.
struct BotQuoteView: View {
    let botName: String
    let quote: Quote
    let onBuy: () -> Void
    let onSell: () -> Void
    let onPass: () -> Void
    var buyEnabled: Bool = true
    var sellEnabled: Bool = true
    var passEnabled: Bool = true

    var body: some View {
        VStack(spacing: 10) {
            // Bot name + quote
            HStack {
                Text(botName)
                    .font(.headline)
                    .foregroundStyle(.white)

                Spacer()

                HStack(spacing: 6) {
                    Text("\(quote.bid)")
                        .foregroundStyle(.red)
                        .fontWeight(.bold)
                    Text("—")
                        .foregroundStyle(.secondary)
                    Text("\(quote.ask)")
                        .foregroundStyle(.green)
                        .fontWeight(.bold)
                }
                .font(.title3.monospacedDigit())
            }

            // Action buttons: Buy / Sell / Pass
            HStack(spacing: 8) {
                Button {
                    onBuy()
                } label: {
                    Text("Buy \(quote.ask)")
                        .font(.subheadline)
                        .fontWeight(.semibold)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 10)
                        .background(buyEnabled ? Color.green.opacity(0.25) : Color.gray.opacity(0.1))
                        .foregroundStyle(buyEnabled ? .green : .gray)
                        .clipShape(RoundedRectangle(cornerRadius: 8))
                }
                .disabled(!buyEnabled)

                Button {
                    onSell()
                } label: {
                    Text("Sell \(quote.bid)")
                        .font(.subheadline)
                        .fontWeight(.semibold)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 10)
                        .background(sellEnabled ? Color.red.opacity(0.25) : Color.gray.opacity(0.1))
                        .foregroundStyle(sellEnabled ? .red : .gray)
                        .clipShape(RoundedRectangle(cornerRadius: 8))
                }
                .disabled(!sellEnabled)

                Button {
                    onPass()
                } label: {
                    Text("Pass")
                        .font(.subheadline)
                        .fontWeight(.semibold)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 10)
                        .background(passEnabled ? Color.white.opacity(0.1) : Color.gray.opacity(0.05))
                        .foregroundStyle(passEnabled ? Color.secondary : Color.gray.opacity(0.5))
                        .clipShape(RoundedRectangle(cornerRadius: 8))
                }
                .disabled(!passEnabled)
            }
        }
        .padding(12)
        .background(Color.white.opacity(0.05))
        .clipShape(RoundedRectangle(cornerRadius: 12))
    }
}

File: kotlin/app/src/main/java/com/magcry/ui/BotQuoteView.kt
package com.magcry.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.magcry.model.Quote

/**
 * Shows a single bot's live quote with Buy / Sell / Pass action buttons.
 * Layout: inline HStack with name left, bid-ask right with em-dash.
 * Transparent button backgrounds matching Swift.
 */
@Composable
fun BotQuoteView(
    botName: String,
    quote: Quote,
    onBuy: () -> Unit,
    onSell: () -> Unit,
    onPass: () -> Unit,
    buyEnabled: Boolean = true,
    sellEnabled: Boolean = true,
    passEnabled: Boolean = true,
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color.White.copy(alpha = 0.05f), RoundedCornerShape(12.dp))
            .padding(12.dp),
        verticalArrangement = Arrangement.spacedBy(10.dp)
    ) {
        // Bot name + quote — inline HStack
        Row(
            modifier = Modifier.fillMaxWidth(),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = botName,
                color = Color.White,
                fontSize = 17.sp,
                fontWeight = FontWeight.Bold
            )
            Spacer(Modifier.weight(1f))
            Row(
                horizontalArrangement = Arrangement.spacedBy(6.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "${quote.bid}",
                    color = Color.Red,
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold,
                    fontFamily = FontFamily.Monospace
                )
                Text(
                    text = "\u2014",  // em-dash
                    color = Color.Gray,
                    fontSize = 18.sp
                )
                Text(
                    text = "${quote.ask}",
                    color = Color.Green,
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold,
                    fontFamily = FontFamily.Monospace
                )
            }
        }

        // Action buttons: Buy / Sell / Pass — transparent backgrounds
        Row(
            horizontalArrangement = Arrangement.spacedBy(8.dp),
            modifier = Modifier.fillMaxWidth()
        ) {
            // Buy button
            Button(
                onClick = onBuy,
                enabled = buyEnabled,
                colors = ButtonDefaults.buttonColors(
                    containerColor = if (buyEnabled) Color.Green.copy(alpha = 0.25f) else Color.Gray.copy(alpha = 0.1f),
                    contentColor = if (buyEnabled) Color.Green else Color.Gray,
                    disabledContainerColor = Color.Gray.copy(alpha = 0.1f),
                    disabledContentColor = Color.Gray,
                ),
                shape = RoundedCornerShape(8.dp),
                contentPadding = PaddingValues(vertical = 10.dp),
                modifier = Modifier.weight(1f)
            ) {
                Text(
                    "Buy ${quote.ask}",
                    fontSize = 14.sp,
                    fontWeight = FontWeight.SemiBold
                )
            }

            // Sell button
            Button(
                onClick = onSell,
                enabled = sellEnabled,
                colors = ButtonDefaults.buttonColors(
                    containerColor = if (sellEnabled) Color.Red.copy(alpha = 0.25f) else Color.Gray.copy(alpha = 0.1f),
                    contentColor = if (sellEnabled) Color.Red else Color.Gray,
                    disabledContainerColor = Color.Gray.copy(alpha = 0.1f),
                    disabledContentColor = Color.Gray,
                ),
                shape = RoundedCornerShape(8.dp),
                contentPadding = PaddingValues(vertical = 10.dp),
                modifier = Modifier.weight(1f)
            ) {
                Text(
                    "Sell ${quote.bid}",
                    fontSize = 14.sp,
                    fontWeight = FontWeight.SemiBold
                )
            }

            // Pass button
            Button(
                onClick = onPass,
                enabled = passEnabled,
                colors = ButtonDefaults.buttonColors(
                    containerColor = if (passEnabled) Color.White.copy(alpha = 0.1f) else Color.Gray.copy(alpha = 0.05f),
                    contentColor = if (passEnabled) Color.Gray else Color.Gray.copy(alpha = 0.5f),
                    disabledContainerColor = Color.Gray.copy(alpha = 0.05f),
                    disabledContentColor = Color.Gray.copy(alpha = 0.5f),
                ),
                shape = RoundedCornerShape(8.dp),
                contentPadding = PaddingValues(vertical = 10.dp),
                modifier = Modifier.weight(1f)
            ) {
                Text(
                    "Pass",
                    fontSize = 14.sp,
                    fontWeight = FontWeight.SemiBold
                )
            }
        }
    }
}
